### PR TITLE
Support for validators advertising their storage last_id

### LINKED ISSUE #2369: Restore validators advertising storage last_id change
#### Problem

Original PR #2154 to allow validators to advertise their storage last_id and implement the storage contract logic broke the test net and was reverted.

#### Proposed Solution

Debug the problem and re-merge the change.

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

Previous version of the change broke the test net.

#### Summary of Changes

Previous change but fixes to avoid testnet breakage.

Fixes #2369

### CODE REVIEW

File: programs/native/storage/src/lib.rs [hunk 1]
```diff
@@ -46,8 +187,41 @@ fn entrypoint(
...
+    fn test_transaction(
+        tx: &Transaction,
+        program_accounts: &mut [Account],
+    ) -> Result<(), ProgramError> {
+        assert_eq!(tx.instructions.len(), 1);
+        let Instruction {
+            ref accounts,
+            ref userdata,
+            ..
+        } = tx.instructions[0];
```
Comment (garious) [inline programs/native/storage/src/lib.rs]: If you borrow `tx.instructions[0]`, you don't need to do that awkward pattern match:

```rust
let ix = &tx.instructions[0];
info!("accounts: {:?}", ix.accounts);
```

File: src/bank.rs [hunk 2]
```diff
@@ -1931,4 +1965,67 @@ mod tests {
...
     }
 
+    #[test]
+    fn test_bank_storage() {
+        solana_logger::setup();
+        let alice = Mint::new(1000);
+        let bank = Bank::new(&alice);
+
+        let bob = Keypair::new();
+        let jack = Keypair::new();
```
Comment (garious) [inline src/bank.rs]: fyi https://en.wikipedia.org/wiki/Alice_and_Bob#Cast_of_characters

**Comment by garious** [comment]
wow, 1000+ LoC PR sleeping under a WIP label. What's going on here?

**Comment by sakridge** [comment]
@garious I find i have to add more code to get it to work... :(

**Comment by sakridge** [comment]
Superseded with some smaller PRs.

**Pull request closed without merging**